src/main/java/Matchers/TagMatcher.java

```diff
@@ -1,10 +1,44 @@
 package main.java.Matchers;
 
+import main.java.EntityInterfaces.RecipeImpl;
+import main.java.Entities.Tag;
+import main.java.Storages.RecipeStorage;
+import main.java.Entities.Recipe;
+import main.java.Entities.Item;
+
+
+
+import java.util.List;
+
 /**
- * Match recipes with given set of tags
+ * Match recipes based on tags
  */
-public class TagMatcher {
+public class TagMatcher implements MatcherImpl {
+    List<Tag> tags;
+
+    public TagMatcher(List<Tag> tags) {
+        this.tags = tags;
+    }
+
+    @Override
+    /**
+     * Checks if item/ingredient contains any of the tags in the matcher
+     */
+    public Boolean matches(Recipe recipe) {
+        for (Tag tag: this.tags) {
+            for (Item item : recipe.items()) {
```

Comment:
Due to the mix up in naming the interfaces, the Recipe and Tag here (and likely elsewhere) are actually referencing the concrete entities, NOT the interfaces.